src/app/lorax/services/share.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @fileOverview ShareService
+ * @author <a href="mailto:chris@work.co">Chris James</a>
+ * @author <a href="mailto:sneethling@mozilla.com">Schalk Neethling</a>
+ */
+define(['angular', 'jquery'], function (angular, $) {
+    'use strict';
+
+    var ShareService = function (
+        dataService,
+        routesService
+    ) {
+
+        this._dataService = dataService;
+        this._routesService = routesService;
+
+        function share(e, service) {
+
+            this._dataService.getMain().then(function (model) {
+                var content = model.getServices().share;
+                var fbShareUrl = content.fbShareUrl;
+                var twitterShareUrl = content.twitterShareUrl;
+                var absUrl = this._routesService._$location.$$absUrl;
+
+                // get the current path from the route service
+                var path = this._routesService._$location.$$path;
+                // get the current issue from the path
+                var issueString = path.substring(path.lastIndexOf('/') + 1);
+                if (issueString) {
+                    this._dataService.getMain().then(function (model) {
+                        var issue = model.getIssueById(issueString);
+
+                        // set base URL based on service
+                        var socialShareUrl = service === 'fb' ? fbShareUrl : twitterShareUrl;
+                        if (service === 'fb') {
+                            socialShareUrl += issue ? issue.getShareUrl() : absUrl;
+                        } else {
+                            socialShareUrl += '?text=';
+                            socialShareUrl += issue ? issue.getTitle() : '';
```

Comment:
I reckon I am just going to replace the &amp; with & and then encode the URL before calling window.open